### PR TITLE
Student answer feedback UI (With gifs)

### PR DESCRIPTION
**Pull request by pamelalozano16**
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the new behavior?

The student selects the answer and when he submits his answer, it is registered in the backend. Then he has to wait until the teacher decides to end the question to see his results.

## Other information
Screenshot 2020-08-12 at 17 11 17 - Display 2
Screenshot 2020-08-14 at 13 59 54 - Display 2
Screenshot 2020-08-14 at 13 59 38 - Display 2

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java [hunk 1]
```diff
@@ -35,7 +35,23 @@
...
+    if(gameInstance==null || gameInstance.isEmpty()){
+        response.setStatus(500);
+        response.getWriter().println("Room not specified");
+        return;
+    }
+
+    String token = request.getParameter("student");
+    if(token==null || token.isEmpty()){
+        response.setStatus(500);
+        response.getWriter().println("Room not specified");
```
Comment (aroquev00) [inline src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:48]: not that it matters a lot, but the response should say something like: user not logged in right? Because the ID token would be empty?

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java [hunk 2]
```diff
@@ -76,10 +92,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DocumentReference questionDocRef = firestoreDb.collection("games").document(gameId).collection("questions")
         .document(questionId);
 
+    boolean isAnswerCorrect;
 
-    String correctAnswerId = getCorrectAnswer(questionId, questionDocRef);
-
-    boolean isAnswerCorrect = answerId.equals(correctAnswerId);
+    //If there was no answer it's wrong
+    if(answerId == null || answerId.isEmpty()){
```
Comment (aroquev00) [inline src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:98]: In which cases can answerId be null and in which case can it be an empty string?

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java [hunk 3]
```diff
@@ -98,10 +120,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       // register the the question's general answer statistics
       updateGeneralQuestionStats(questionInGameInstanceDocRef, isAnswerCorrect, firestoreDb);
-
+      if(answerId != null && !answerId.isEmpty()){
```
Comment (aroquev00) [inline src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:123]: I'm thinking that maybe we should make sure that an answerId cannot be empty when it is sent to the servlet? Or if it is possible, don't count it at all. Because if it is empty, we are already registering that the student got it wrong in his record (call in line 109), but we are not saving this "question answer" in the general statistics for the question, so this could lead to data inconsistency in the DB.

File: src/main/webapp/student/scripts/play-game.js [hunk 4]
```diff
@@ -58,10 +61,20 @@ function getGameInstanceIdFromQueryParams() {
...
+  isCorrect = null;
+  document.getElementById("gif").setAttribute('src','')
+}
+
 // Inits listener to User's points in Firestore DB
-function initPointsListener(gameInstanceId, user) {
-  db.collection("gameInstance").doc(gameInstanceId).collection("students").doc(user.uid)
-  .onSnapshot(function(doc) {
+function updatePoints() {
+  db.collection("gameInstance").doc(gameInstanceId).collection("students").doc(studentId.uid).get().then(function(doc) {
```
Comment (aroquev00) [inline src/main/webapp/student/scripts/play-game.js:77]: Well thought for the points! Nice that now its queried when the question is finished and not before.

File: src/main/webapp/student/scripts/play-game.js [hunk 5]
```diff
@@ -147,16 +162,34 @@ function initSubmitButton(gameInstanceId, gameId) {
...
 
   // Add the question title to the UI
   createQuestionObject(currentQuestion.title);
 
+  if(!currentQuestionActive){
+      if(!submited){
+        document.getElementById('submitButton').disabled = false;
+        document.getElementById("submitButton").click();
+        document.getElementById('submitButton').disabled = true;
+        document.getElementById("quiz").innerHTML='';
```
Comment (aroquev00) [inline src/main/webapp/student/scripts/play-game.js:184]: What is this for? An empty answer is being sent right?

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java (hunk 2 above)
Comment (aroquev00) [reply src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:98]: Oh I think I know why, if the user runs out of time then the answer would be an empty string right?

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java (hunk 3 above)
Comment (aroquev00) [inline src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:123]: Well thought for empty answers!

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java (hunk 1 above)
Comment (pamelalozano16) [reply src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:48]: You are right thank you!

File: src/main/webapp/student/scripts/play-game.js (hunk 5 above)
Comment (pamelalozano16) [reply src/main/webapp/student/scripts/play-game.js:184]: Yeah so this means the current question is not active and there wasn't a submission or selection so it is submitted automatically, for example, in case that the time is up and the teacher ends the question, if the student didn't submit an answer it submits null which returns a wrong answer.

File: src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java (hunk 2 above)
Comment (pamelalozano16) [reply src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java:98]: Exactly

File: src/main/webapp/student/play-game.html [hunk 6]
```diff
@@ -54,6 +54,10 @@
   <link rel="stylesheet" href="../styles/main.css">
   <link rel="stylesheet" href="styles/play-game.css">
 
+  <script
+  src="https://code.jquery.com/jquery-3.5.1.js"
+  integrity="sha256-QWo7LDvxbWT2tbbQ97B53yJnYU3WhH/C8ycbRAkjPDc="
+  crossorigin="anonymous"></script>
```
Comment (thefredo1000) [inline src/main/webapp/student/play-game.html:60]: Why did you add JQuery?

File: src/main/webapp/student/scripts/play-game.js [hunk 7]
```diff
@@ -29,6 +35,7 @@ function initAuthStateObserver() {
 function authStateObserver(user) {
   if (user) { // User is signed in!
     // Everything starts working when the User logs in
+    studentId = user;
     loadGamePanel(user)
   } else { // User is signed out!
     console.log("Not logged in");
```
Comment (thefredo1000) [inline src/main/webapp/student/scripts/play-game.js:41]: Should we add something to handle this?

Comment (thefredo1000) [reply src/main/webapp/student/scripts/play-game.js:41]: Or maybe it not necessary

File: src/main/webapp/student/scripts/play-game.js [hunk 8]
```diff
@@ -242,5 +278,47 @@ function createAnswer(quiz, multipleDiv, doc, i){
...
+        const info = await infoJson.json(); 
+        isCorrect = info.correct;
+        resultObject.classList.toggle(isCorrect.toString());
+        if(isCorrect) {
+            tag = "celebrate";
+            resultObject.innerText = 'Correct!';
+        } else {
+            tag = "disappointment";
+            resultObject.innerText = 'Incorrect :(';
+        }
```
Comment (thefredo1000) [inline src/main/webapp/student/scripts/play-game.js:297]: Nit: We can add the word "Disney" to the tag, so even if it has a hard time filtering some stuff, we know that it's really family friendly

File: src/main/webapp/student/play-game.html (hunk 6 above)
Comment (pamelalozano16) [reply src/main/webapp/student/play-game.html:60]: Oh I was going to use it but let me delete it, thanks!

File: src/main/webapp/student/scripts/play-game.js (hunk 8 above)
Comment (pamelalozano16) [reply src/main/webapp/student/scripts/play-game.js:297]: Right
